### PR TITLE
Select inserted space after join

### PR DESCRIPTION
**Pull request by pickfire**
[asciicast](https://asciinema.org/a/Y5Jq6mJHg41UUuZeInU4jfJGw)

Will be useful to remove those inserted space if I want.

### CODE REVIEW

**Comment by the-mikedavis** [comment]
I think in kakoune this ends up being a different command - `A-J` if I remember correctly?

**Comment by EpocSquadron** [comment]
Yes and it should remain a separate command as you don't want to modify the users selection except as opt-in, say if you have a range inclusive of many newlines and want to join them internally, you may not want to lose the overall selection.

File: book/src/keymap.md [hunk 1]
```diff
@@ -129,6 +129,7 @@
 | `X`                   | Extend selection to line bounds (line-wise selection)             | `extend_to_line_bounds`              |
 | `Alt-x`               | Shrink selection to line bounds (line-wise selection)             | `shrink_to_line_bounds`              |
 | `J`                   | Join lines inside selection                                       | `join_selections`                    |
+| `A-J`                 | Join lines inside selection and select space                      | `join_selections_space`              |
```
Comment (pickfire) [inline book/src/keymap.md:132]: I am not sure if we want to use `A-j` or `A-J` so I just go with `A-J`.

Comment (Omnikar) [reply book/src/keymap.md:132]: `A-J` makes more sense since it's an alternate behaviour of `J`.

**Pull request merged**